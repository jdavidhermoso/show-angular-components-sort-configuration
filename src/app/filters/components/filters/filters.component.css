.filters__container {
  display: flex;
  flex-direction: column;
}

.filter {
  display: block;
  border: 1px solid;
}
