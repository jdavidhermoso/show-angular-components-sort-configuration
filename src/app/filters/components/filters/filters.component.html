<ng-container *ngIf="(filtersOrderConfiguration$ | async) as orderConfiguration">
  <div class="filters__container">
    <app-filters-difficult
      class="filter"
      [style.order]="orderConfiguration.difficultLevel"
    ></app-filters-difficult>
    <app-filters-price
      class="filter"
      [style.order]="orderConfiguration.price"
    ></app-filters-price>
    <app-filters-duration
      class="filter"
      [style.order]="orderConfiguration.duration"
    ></app-filters-duration>
    <app-filters-rating
      class="filter"
      [style.order]="orderConfiguration.rating"
    ></app-filters-rating>
    <app-filters-language
      class="filter"
      [style.order]="orderConfiguration.language"
    ></app-filters-language>
  </div>
</ng-container>

NOTE: This component is just to allow you see how the components change! It would not be present in the real solution.
<app-filters-form (filtersUpdate)="updateFiltersOrder($event)"></app-filters-form>


