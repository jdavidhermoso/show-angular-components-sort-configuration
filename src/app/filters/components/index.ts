export * from './filters-difficult';
export * from './filters-language';
export * from './filters-rating';
export * from './filters-price';
export * from './filters-duration';
export * from './filters-form';
export * from './filters';
