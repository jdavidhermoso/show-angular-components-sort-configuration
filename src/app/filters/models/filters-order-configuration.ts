export interface FiltersOrderConfiguration {
  price: number;
  language: number;
  rating: number;
  duration: number;
  difficultLevel: number;
}
