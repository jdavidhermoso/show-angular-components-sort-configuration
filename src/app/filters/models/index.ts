export * from './filters-order-configuration';
