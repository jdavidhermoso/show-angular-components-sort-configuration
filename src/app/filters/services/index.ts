export * from './demo.service';
export * from './filters.service';
