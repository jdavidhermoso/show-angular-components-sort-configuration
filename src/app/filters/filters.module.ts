import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {FiltersPriceComponent} from './components/filters-price/filters-price.component';
import {FiltersLanguageComponent} from './components/filters-language/filters-language.component';
import {FiltersRatingComponent} from './components/filters-rating/filters-rating.component';
import {FiltersDurationComponent} from './components/filters-duration/filters-duration.component';
import {FiltersDifficultComponent} from './components/filters-difficult/filters-difficult.component';
import {FiltersComponent} from './components/filters/filters.component';
import {FiltersFormComponent} from '../components/filters-form/filters-form.component';
import {ReactiveFormsModule} from '@angular/forms';


@NgModule({
  declarations: [
    FiltersPriceComponent,
    FiltersLanguageComponent,
    FiltersRatingComponent,
    FiltersDurationComponent,
    FiltersDifficultComponent,
    FiltersComponent,
    FiltersFormComponent
  ],
  exports: [
    FiltersComponent,
  ],
  imports: [
    CommonModule,
    ReactiveFormsModule
  ]
})
export class FiltersModule { }
