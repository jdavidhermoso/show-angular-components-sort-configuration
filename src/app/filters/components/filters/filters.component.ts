import {Component} from '@angular/core';
import {DemoService} from '../../services/demo.service';
import {FiltersOrderConfiguration} from '../../models/filters-order-configuration';
import {Observable} from 'rxjs';
import {FiltersService} from '../../services/filters.service';

@Component({
  selector: 'app-filters',
  templateUrl: './filters.component.html',
  styleUrls: ['./filters.component.css']
})
export class FiltersComponent {
  public filtersOrderConfiguration$: Observable<FiltersOrderConfiguration>;

  constructor(private filtersService: FiltersService,
              private demoService: DemoService) {
    this.filtersOrderConfiguration$ = this.filtersService.getFiltersOrderConfiguration();
  }

  public updateFiltersOrder(filters: FiltersOrderConfiguration) {
    this.demoService.updateFilterOrder(filters);
  }

}
