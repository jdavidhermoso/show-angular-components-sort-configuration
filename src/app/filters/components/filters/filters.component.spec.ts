import {FiltersComponent} from './filters.component';
import {BehaviorSubject, of} from 'rxjs';
import {DemoService} from '../../services';
import {FiltersOrderConfiguration} from '../../models';

describe('FiltersComponent', () => {
  let component: FiltersComponent;
  const demoServiceMock = {
    filtersOrderConfigurationBehaviorSubject: new BehaviorSubject({
      price: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION,
      difficultLevel: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION,
      language: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION,
      rating: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION,
      duration: DemoService.DEFAULT_FILTER_ORDER_CONFIGURATION
    }),
    updateFilterOrder: jest.fn()
  };

  const filtersServiceMock = {
    getFiltersOrderConfiguration: jest.fn(() => {
      return of({
        price: 3,
        difficultLevel: 4,
        language: 2,
        rating: 1,
        duration: 1
      });
    })
  };

  it('should get filtersOrderConfiguration', (done) => {
    component = new FiltersComponent(filtersServiceMock as any, demoServiceMock as any);
    component.filtersOrderConfiguration$.subscribe((filtersOrderConfiguration: FiltersOrderConfiguration) => {
      expect(filtersOrderConfiguration).toMatchSnapshot();
      done();
    });
  });

  it('should call demoService to update filters configuration', () => {
    component = new FiltersComponent(filtersServiceMock as any, demoServiceMock as any);

    component.updateFiltersOrder({
      price: 3,
      difficultLevel: 4,
      language: 2,
      rating: 1,
      duration: 1
    });

    expect(demoServiceMock.updateFilterOrder).toHaveBeenCalled();
  });
});
