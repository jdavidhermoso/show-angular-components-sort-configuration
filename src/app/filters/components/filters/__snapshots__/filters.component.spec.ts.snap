// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersComponent should get filtersOrderConfiguration 1`] = `
Object {
  "difficultLevel": 4,
  "duration": 1,
  "language": 2,
  "price": 3,
  "rating": 1,
}
`;
